### ISSUE TITLE
Is there any way to get concurrent access to the wav file?

### ISSUE THREAD

**Issue by mithunvs** [filoe/cscore#355]
Im new to cscore audio library. Im trying to write a wave file while recording in the wasapi capture data available event, at the same time i need the written data to post to an API in fixed intervals.I want to get access to the file being written but if i dispose the file it will not write in the same fileand if i dont dispose the file i cant use the file to post to the API .Please help..

**Comment by filoe** [comment]
So you want to save the audio data in a file AND post it to an api at the same time?

**Comment by mithunvs** [comment]
Is there any stream wrapper like Ignore dispose stream or something like that

**Comment by filoe** [comment]
The source above will do that. As soon as a block of 5 seconds was processed, it will provide it to you.
Use e.Data to get the read samples.

**Comment by mithunvs** [comment]
I used a code like this because the e.data gives float values but the wave writer only writes the byte values and still i get a  1 second of data,can u please suggest the code to get a 5 second data
`notificationSource.BlockRead += (s, e) =>
			{
				float[] interceptsBuff = new float[notificationSource.WaveFormat.BytesPerSecond / 2];
				
				string directory = System.IO.Directory.GetCurrentDirectory();
				string filename = /*DateTime.Now.Ticks.ToString()*/ "sample"+ ".wav";
				directory = directory + "\\Cache\\";
				string fileName = System.IO.Path.Combine(directory + filename);
				var intercepts = new WaveWriter(fileName, notificationSource.WaveFormat);
				interceptsBuff = e.Data;
				byte[] buff= new byte[interceptsBuff.Length * 4];
				Buffer.BlockCopy(interceptsBuff, 0, buff, 0, buff.Length);
				intercepts.Write(buff, 0, e.Length);
				intercepts.Dispose();


			};`

**Comment by mithunvs** [comment]
`<addr>`

**Comment by filoe** [comment]
So you want to create a new wave file every few seconds? 
Why would you want to do that? To post it to an api? Why don't you just stream the byte data to the api?

**Comment by mithunvs** [comment]
Yes i want to create a file every few seconds. I need this 5 seconds data for later use. Is there any way for doing that.

**Comment by mithunvs** [comment]
Thanks that's what I've been looking for. Works like a charm.Now it records and gives data at the specified interval but what if i need the data if a particular event triggers.Can you help me for this last time
